meta-mel-support/recipes-core/plymouth/plymouth-mel/mel.script

```diff
@@ -0,0 +1,162 @@
+bg_image = Image("special://logo");
+
+screen_ratio = Window.GetHeight() / Window.GetWidth();
+bg_image_ratio = bg_image.GetHeight() / bg_image.GetWidth();
+
+if (screen_ratio < bg_image_ratio)
+  {  # Screen ratio is taller than image ratio, we will match the screen height
+     scale_factor =  Window.GetHeight() / bg_image.GetHeight();
+  }
+else
+  {  # Screen ratio is wider than image ratio, we will match the screen width
+     scale_factor =  Window.GetWidth() / bg_image.GetWidth();
+  }
+#scale_factor = Window.GetWidth() / bg_image.GetWidth();
+
+scaled_bg_image = bg_image.Scale(bg_image.GetWidth()  * scale_factor,
+                                 bg_image.GetHeight() * scale_factor);
```

Comment:
That'll distort the image when the aspect ratio of the image isn't the same as the aspect ratio of the display, since the width and height factors wouldn't be the same in that case.